### PR TITLE
fix(react): export types for Stack component

### PR DESCRIPTION
**Pull request by onurozkardes**
Ref: https://github.com/carbon-design-system/carbon/issues/13577
Follow up to https://github.com/carbon-design-system/carbon/pull/15404

Add some missing typescript exports for Stack component.

#### Changelog

**New**

- {{new thing}}

**Changed**

- Add missing Stack component export as default.
- Add missing StackProps interface export.
- Isolate & import type declarations of Stack component.
- Isolate & import definitions (SPACING_STEPS) of Stack component.

**Removed**

- Component type declarations (StackProps) moved to new "types.ts" file from "Stack.tsx" file. In this way, complicated code block has been prevented. 

#### Testing / Reviewing

- Compiled component files have been tested on storybook playground. 
- Compiled component files have been tested on a project which works with **Carbon Design System**.

### CODE REVIEW

**Comment by onurozkardes** [comment]
I have read the DCO document and I hereby sign the DCO.

**Comment by onurozkardes** [comment]
> **DCO Assistant Lite bot:** Thanks for your submission! We ask that you sign our [Developer Certificate of Origin](https://github.com/carbon-design-system/carbon-dco/blob/main/dco.md) before we can accept your contribution. You can sign the DCO by adding a comment below using this text:
> 
> I have read the DCO document and I hereby sign the DCO.
> 
> **Onur Özkardeş** seems not to be a GitHub user. You need a GitHub account to be able to sign the DCO. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).You can retrigger this bot by commenting **recheck** in this Pull Request

I have read the DCO document and I hereby sign the DCO.

**Comment by onurozkardes** [comment]
I have read the DCO document and I hereby sign the DCO.

**Review by andreancardona — APPROVED** [review]

File: packages/react/src/components/Stack/types.ts
Comment (tay1orjones) [inline packages/react/src/components/Stack/types.ts:1]: So far the convention we've been following is to have the interface directly in the component source file.

File: packages/react/src/components/Stack/definitions.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+/**
+ * The steps in the spacing scale
+ * @type {Array<number>}
+ */
+export const SPACING_STEPS = Array.from({ length: spacing.length - 1 }).map(
+  (_, step) => {
+    return step + 1;
+  }
+);
```
Comment (tay1orjones) [inline packages/react/src/components/Stack/definitions.ts]: To my knowledge this does not (and likely will never) be used outside of the context of `Stack`. Is there a reason requiring this to be in a separate file?

File: packages/react/src/components/Stack/Stack.tsx [hunk 2]
```diff
@@ -133,4 +93,4 @@ Stack.propTypes = {
   orientation: PropTypes.oneOf(['horizontal', 'vertical']),
 };
 
-export { Stack };
+export default Stack;
```
Comment (tay1orjones) [inline packages/react/src/components/Stack/Stack.tsx]: Could you elaborate on why the named export is a problem?

**Review by tay1orjones — CHANGES_REQUESTED** [review]
Hey there, thanks for this. Could you outline further what the problem is that this is solving?

There's a few conventions we've been following that I outlined below.

**Comment by onurozkardes** [comment]
> Hey there, thanks for this. Could you outline further what the problem is that this is solving?
> 
> There's a few conventions we've been following that I outlined below.

Hey there,

I'm glad to your review. 
Firstly, I had faced a issue when I used the component on my React & Typescript project due to lack of type declarations on Stack component.
I had purposed to fix that issue and improve code readability by dividing some files (like types, definitions) at my first commit.
Then, as you say, there were some changes that out of your standard. 

I refactored some parts of the code below that you've outlined.

- Type declarations has been moved into main component file again.
- "SPACING_STEPS" variable has been moved into main component file again. 
- "HStack" and "VStack" functions have been moved to its own files. 
- "index.ts" has been changed name instead of "index.tsx". 
- Component exportations has been revised according to what you outlined.

As far as I see, other component index.ts files look quite simple, that's why I divided "HStack" and "VStack" components into new files from "index.tsx".
In this way, "index.ts" file looks very simple anymore and I've just exported related files & types.  

Best regards.

File: packages/react/src/components/Stack/Stack.tsx [hunk 3]
```diff
@@ -5,23 +5,24 @@
...
-const SPACING_STEPS = Array.from({ length: spacing.length - 1 }).map(
-  (_, step) => {
-    return step + 1;
-  }
-);
+export const SPACING_STEPS = Array.from({
+  length: TextLineSpacing.length - 1,
+}).map((_, step) => {
+  return step + 1;
+});
```
Comment (tay1orjones) [inline packages/react/src/components/Stack/Stack.tsx]: This should be reverted. Why would the `TextLineSpacing` icon from `@carbon/icons-react` be used to generate the valid spacing steps?

image

**Review by tay1orjones — CHANGES_REQUESTED** [review]
Getting close on this one, thanks for your continued effort here

**Comment by onurozkardes** [comment]
> Getting close on this one, thanks for your continued effort here

Hi there,

Thanks for your feedback.
I've just fixed that.

**Review by tay1orjones — APPROVED** [review]

**Comment by tay1orjones** [comment]
@onurozkardes sorry for the delay here - this has a few conflicts that need resolved

**Review by andreancardona — APPROVED** [review]
tested and looks good!

**Pull request merged**